src/ansiblelint/utils.py

```diff
@@ -521,60 +521,83 @@ def _sanitize_task(task: Dict[str, Any]) -> Dict[str, Any]:
     return result
 
 
-def normalize_task_v2(task: Dict[str, Any]) -> Dict[str, Any]:
-    """Ensure tasks have a normalized action key and strings are converted to python objects."""
+def _get_dict_with_existing_key(task: Dict[str, Any]) -> Dict[str, Any]:
+    """Return a dict with existing key in task."""
     result = {}
-
-    sanitized_task = _sanitize_task(task)
-    mod_arg_parser = ModuleArgsParser(sanitized_task)
-    try:
-        action, arguments, result["delegate_to"] = mod_arg_parser.parse(
-            skip_action_validation=options.skip_action_validation
-        )
-    except AnsibleParserError as exc:
-        # pylint: disable=raise-missing-from
-        raise MatchError(
-            rule=AnsibleParserErrorRule(),
-            message=exc.message,
-            filename=task.get(FILENAME_KEY, "Unknown"),
-            linenumber=task.get(LINE_NUMBER_KEY, 0),
-        )
-
-    # denormalize shell -> command conversion
-    if "_uses_shell" in arguments:
-        action = "shell"
-        del arguments["_uses_shell"]
-
     for (k, v) in list(task.items()):
-        if k in ("action", "local_action", "args", "delegate_to") or k == action:
+        if k in ("action", "local_action", "args", "delegate_to"):
             # we don't want to re-assign these values, which were
             # determined by the ModuleArgsParser() above
             continue
         result[k] = v
 
-    if not isinstance(action, str):
-        raise RuntimeError(f"Task actions can only be strings, got {action}")
-    action_unnormalized = action
-    # convert builtin fqn calls to short forms because most rules know only
-    # about short calls but in the future we may switch the normalization to do
-    # the opposite. Mainly we currently consider normalized the module listing
-    # used by `ansible-doc -t module -l 2>/dev/null`
-    action = removeprefix(action, "ansible.builtin.")
-    result["action"] = dict(
-        __ansible_module__=action, __ansible_module_original__=action_unnormalized
-    )
+    return result
 
-    if "_raw_params" in arguments:
-        result["action"]["__ansible_arguments__"] = arguments["_raw_params"].split(" ")
-        del arguments["_raw_params"]
+
+def normalize_task_v2(task: Dict[str, Any]) -> Dict[str, Any]:
+    """Ensure tasks have a normalized action key and strings are converted to python objects."""
+    if include_nested_task_key(task):
+        result = _get_dict_with_existing_key(task)
+        # Add dummy action for block/always/rescue statements
+        result["action"] = dict(
+            __ansible_module__="block/always/rescue",
+            __ansible_module_original__="block/always/rescue",
+        )
```

Comment:
The GitHub UI makes reviewing the rest of this function more difficult as it looks a bit messy.

It looks like this block could return early:
```suggestion
        result["action"] = dict(
            __ansible_module__="block/always/rescue",
            __ansible_module_original__="block/always/rescue",
        )
        return result
```


Returning early allows dedenting the rest of this function which should minimize the diff and make it easier to review the rest of this. Wdyt? Would an early return work here, or did I miss something?

Reply:
Thanks, it works here so I added and refactored this part in the commit [fff8973](https://github.com/ansible/ansible-lint/pull/2031/commits/fff89733875d05232a128aef7fd3381bbe3d58ab).